internal/provider/catalog/media_datasource.go

```diff
@@ -0,0 +1,171 @@
+// Package catalog provides a Terraform datasource.
+package catalog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/client"
+)
+
+var (
+	_ datasource.DataSource              = &catalogMediaDataSource{}
+	_ datasource.DataSourceWithConfigure = &catalogMediaDataSource{}
+)
+
+func NewCatalogMediaDataSource() datasource.DataSource {
+	return &catalogMediaDataSource{}
+}
+
+type catalogMediaDataSource struct {
+	client *client.CloudAvenue
+}
+
+type catalogMediaDataSourceModel struct {
+	ID             types.String `tfsdk:"id"`
+	CatalogID      types.String `tfsdk:"catalog_id"`
+	Name           types.String `tfsdk:"name"`
+	Description    types.String `tfsdk:"description"`
+	IsISO          types.Bool   `tfsdk:"is_iso"`
+	OwnerName      types.String `tfsdk:"owner_name"`
+	IsPublished    types.Bool   `tfsdk:"is_published"`
+	CreatedAt      types.String `tfsdk:"created_at"`
+	Size           types.Int64  `tfsdk:"size"`
+	Status         types.String `tfsdk:"status"`
+	StorageProfile types.String `tfsdk:"storage_profile"`
+}
+
+func (d *catalogMediaDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + categoryName + "_" + "media"
+}
+
+func (d *catalogMediaDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "The catalog_media datasource provides a VMware Cloud Director Catalog media data source.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The ID of the catalog media.",
+			},
+			"catalog_id": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The ID of the catalog to which media file belongs.",
+			},
```

Comment:
I suggest to add optional `catalog_name` to let the user choose between `catalog_name` or `catalog_id`